Doc: update 'Configure Grafana Mimir zone-aware replication' page (#13086)
grafana/mimir: docs/sources/mimir/configure/configure-zone-aware-replication.md

## Configuring ingester time series replication

Mimir replicates time series across ingesters in fundamentally different ways between the ingest-storage and classic architectures.

### Configuring ingester time series replication in ingest storage architecture

{{< admonition type="note" >}}
This guidance applies to ingest storage architecture. For more information about the supported architectures in Grafana Mimir, refer to [Grafana Mimir architecture](https://grafana.com/docs/mimir/<MIMIR_VERSION>/get-started/about-grafana-mimir-architecture/).
{{< /admonition >}}

In the ingest storage architecture, distributors shard incoming series across Kafka partitions, writing each series to a single partition.
Series are not replicated across partitions, since Kafka itself provides replication within each partition.

Each ingester owns one Kafka partition and continuously consumes series data from it, making that data available for querying.
A partition can be assigned to multiple ingesters for high availability – typically two ingesters per partition, corresponding to a replication factor of two.

> In this architecture, no explicit replication factor needs to be configured.

Reviewer: ```suggestion
In this architecture, you don't need to configure an explicit replication factor.
```

2. Roll out ingesters so that each ingester replica runs with a configured zone.
3. Set the `-ingester.ring.zone-awareness-enabled=true` CLI flag or its respective YAML configuration parameter for distributors, ingesters, and queriers.

### Configuring ingester time series replication in classic architecture

{{< admonition type="note" >}}
This guidance applies to classic architecture. For more information about the supported architectures in Grafana Mimir, refer to [Grafana Mimir architecture](https://grafana.com/docs/mimir/<MIMIR_VERSION>/get-started/about-grafana-mimir-architecture/).
{{< /admonition >}}

In the classic architecture, distributors shard and replicate incoming series across ingesters, writing each series to the number of ingesters defined by `-ingester.ring.replication-factor`.
When zone-aware replication is enabled, each time series is replicated to ingesters located in different zones.

To enable zone-aware replication for time series:

> 1. Configure the zone of each ingester replica via the `-ingester.ring.instance-availability-zone` CLI flag, and the replication factor via the `-ingester.ring.replication-factor` CLI flag, or their respective YAML configuration parameters.

Reviewer: ```suggestion
1. Configure the zone of each ingester replica via the `-ingester.ring.instance-availability-zone` CLI flag and the replication factor via the `-ingester.ring.replication-factor` CLI flag. You can also set their respective YAML configuration parameters.
```
